tikapy/__init__.py

```diff
@@ -241,19 +257,29 @@ def __init__(self, address, port=8729, verify_cert=True,
         self.verify_cert = verify_cert
         self.verify_addr = verify_addr
 
-    def _connect(self):
+    def _connect(self, timeOut):
         """
         Connects a ssl socket.
+
+        :param timeOut: Time set for the timeout for the API connections
+        attempt.
         """
-        self._connect_socket()
+        self._connect_socket(timeOut)
         try:
             ctx = ssl.create_default_context()
             if not self.verify_cert:
                 ctx.verify_mode = ssl.CERT_OPTIONAL
             if not self.verify_addr:
                 ctx.check_hostname = False
-            self._sock = ctx.wrap_socket(self._base_sock,
-                                         server_hostname=self.address)
+            
+            ## Added due to SSLv3 errors happening. This is even though ssl.create_default_context()
+            ## is suppose to set OP_NO_SSLv3, but it still tries to use SSLv3 and subsequently gets an
+            ## error. This was found on Windows and Linux environments. To bypass this you require to 
+            ## use ADH as the cipher with SECLEVEL set to 0.
+            ctx.set_ciphers('ADH:@SECLEVEL=0')
```

Comment:
This seems really dubious.

Is it necessary for connecting to routers which don't have any certificate configured? If so, I think it should require explicit opt-in (similar to existing `allow_insecure_...`).